### PR TITLE
IGNITE-19955 Unmute some tests for DistributionZoneManager

### PR DESCRIPTION
**Pull request by alievmirza**
https://issues.apache.org/jira/browse/IGNITE-19955

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/causalitydatanodes/CausalityDataNodesEngine.java [hunk 1]
```diff
@@ -173,16 +173,22 @@ public CompletableFuture<Set<String>> dataNodes(long causalityToken, int catalog
...
             ) {
-                // It means that the zone was created but the data nodes value had not updated yet.
+                // It means that the zone was created but the data nodes value had not been updated yet.
                 // So the data nodes value will be equals to the logical topology on the descLastUpdateRevision.
+                Entry topologyEntry = msManager.getLocally(zonesLogicalTopologyKey(), createRevision);
 
-                Entry topologyEntry = msManager.getLocally(zonesLogicalTopologyKey(), descLastUpdateRevision);
-
-                assert topologyEntry.value() != null : "Logical topology must be initialized.";
+                if (topologyEntry.empty()) {
```
[the remark this answers is not in the document]
Comment (alievmirza) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/causalitydatanodes/CausalityDataNodesEngine.java:187]: added `testEmptyDataNodesOnZoneCreationBeforeTopologyEvent`

File: modules/distribution-zones/src/integrationTest/java/org/apache/ignite/internal/distributionzones/ItIgniteDistributionZoneManagerNodeRestartTest.java [hunk 2]
```diff
@@ -671,22 +664,28 @@ public void testScaleUpTimerIsRestoredAfterRestart(String zoneName) throws Excep
...
+                catalogManager::latestCatalogVersion,
+                zoneId,
+                Set.of(A, B),
                 TIMEOUT_MILLIS
         );
 
         node.stop();
 
-        startScaleUpBlocking = false;
+        scaleUpLatch.countDown();
```
[the remark this answers is not in the document]
Comment (alievmirza) [reply modules/distribution-zones/src/integrationTest/java/org/apache/ignite/internal/distributionzones/ItIgniteDistributionZoneManagerNodeRestartTest.java:688]: I would prefer to save, so we prevent any potential hanging tasks

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/causalitydatanodes/DistributionZoneCausalityDataNodesTest.java [hunk 3]
```diff
@@ -408,6 +409,92 @@ void dataNodesUpdatedAfterScaleUpChanged() throws Exception {
...
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    void testZoneWithNonImmediateTimersOnCreation() throws Exception {
+        createZone(ZONE_NAME, 1, 1, null);
+
+        putNodeInLogicalTopologyAndGetRevision(NODE_0, ONE_NODE);
+        putNodeInLogicalTopologyAndGetRevision(NODE_1, TWO_NODES);
+        removeNodeInLogicalTopologyAndGetRevision(Set.of(NODE_1), ONE_NODE);
```
[the remark this answers is not in the document]
Comment (alievmirza) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/causalitydatanodes/DistributionZoneCausalityDataNodesTest.java]: done

File: modules/distribution-zones/src/integrationTest/java/org/apache/ignite/internal/distributionzones/ItIgniteDistributionZoneManagerNodeRestartTest.java [hunk 4]
```diff
@@ -972,4 +850,53 @@ private static void alterZone(
...
+        distributionZoneManager.zonesState().get(zoneId)
+                .rescheduleScaleUp(IMMEDIATE_TIMER_VALUE, dummyScaleUpTask, zoneId);
+
+        return scaleUpLatch;
+    }
+
+    /**
+     * Schedule a down up task which block execution of another scale down tasks.
+     * Note that because of the executor of scale up/scale down taksks is single-threaded, this
+     * method also blocks scale down tasks execution.
```
[the remark this answers is not in the document]
Comment (alievmirza) [reply modules/distribution-zones/src/integrationTest/java/org/apache/ignite/internal/distributionzones/ItIgniteDistributionZoneManagerNodeRestartTest.java]: fixed

File: modules/distribution-zones/src/integrationTest/java/org/apache/ignite/internal/distributionzones/ItIgniteDistributionZoneManagerNodeRestartTest.java [hunk 5]
```diff
@@ -972,4 +850,53 @@ private static void alterZone(
...
+        };
+        distributionZoneManager.zonesState().get(zoneId)
+                .rescheduleScaleUp(IMMEDIATE_TIMER_VALUE, dummyScaleUpTask, zoneId);
+
+        return scaleUpLatch;
+    }
+
+    /**
+     * Schedule a down up task which block execution of another scale down tasks.
+     * Note that because of the executor of scale up/scale down taksks is single-threaded, this
```
[the remark this answers is not in the document]
Comment (alievmirza) [reply modules/distribution-zones/src/integrationTest/java/org/apache/ignite/internal/distributionzones/ItIgniteDistributionZoneManagerNodeRestartTest.java]: fixed

File: modules/distribution-zones/src/integrationTest/java/org/apache/ignite/internal/distributionzones/ItIgniteDistributionZoneManagerNodeRestartTest.java [hunk 6]
```diff
@@ -972,4 +850,53 @@ private static void alterZone(
     ) {
         DistributionZonesTestUtil.alterZone(getCatalogManager(node), zoneName, scaleUp, scaleDown, filter);
     }
+
+    /**
+     * Schedule a scale up task which block execution of another scale up tasks.
+     * Note that because of the executor of scale up/scale down taksks is single-threaded, this
```
[the remark this answers is not in the document]
Comment (alievmirza) [reply modules/distribution-zones/src/integrationTest/java/org/apache/ignite/internal/distributionzones/ItIgniteDistributionZoneManagerNodeRestartTest.java]: fixed

File: modules/distribution-zones/src/integrationTest/java/org/apache/ignite/internal/distributionzones/ItIgniteDistributionZoneManagerNodeRestartTest.java [hunk 7]
```diff
@@ -972,4 +850,53 @@ private static void alterZone(
...
+
+        return scaleUpLatch;
+    }
+
+    /**
+     * Schedule a scale down task which block execution of another scale down tasks.
+     * Note that because of the executor of scale up/scale down tasks is single-threaded, this
+     * method also blocks scale up tasks execution.
+     *
+     * @return Latch to unblock execution of scale up tasks.
```
Comment (sergeyuttsel) [inline modules/distribution-zones/src/integrationTest/java/org/apache/ignite/internal/distributionzones/ItIgniteDistributionZoneManagerNodeRestartTest.java:883]: "scale up" -> "scale down"

**Review by sergeyuttsel — APPROVED** [review]

**Pull request merged**